src/goto-programs/goto_convert_side_effect.cpp

```diff
@@ -314,7 +314,13 @@ void goto_convertt::remove_post(
   if(result_is_used)
   {
     exprt tmp = op;
-    make_temp_symbol(tmp, "post", dest, mode);
+    std::string suffix = "post";
+    if (tmp.id() == ID_symbol)
+    {
+      suffix = suffix + "_" + tmp.get(ID_identifier).c_str();
+    }
```

Comment:
I'd like to suggest code along the following lines:
```
if(auto sym_expr = expr_try_dynamic_cast<symbol_exprt>(tmp))
{
  const irep_idt &base_name = ns.lookup(sym_expr->get_identifier()).base_name;
  suffix += "_" + id2string(base_name);
}
```
This will avoid the double colons in the resulting name.